lib/src/signed_video_h26x_sign.c

```diff
@@ -625,30 +621,186 @@ signed_video_add_nalu_part_for_signing_with_timestamp(signed_video_t *self,
+          signature_info->max_signature_size, &signature_info->signature_size, &signature_error)) {
+        SVI_THROW(sv_rc_to_svi_rc(signature_error));
+#ifdef SIGNED_VIDEO_DEBUG
+        // TODO: This might not work for blocked signatures, that is if the hash in
+        // |signature_info| does not correspond to the copied |signature|.
+        // Verify the just signed hash.
+        int verified = -1;
+        SVI_THROW_WITH_MSG(sv_rc_to_svi_rc(openssl_verify_hash(signature_info, &verified)),
+            "Verification test had errors");
+        SVI_THROW_IF_WITH_MSG(verified != 1, SVI_EXTERNAL_FAILURE, "Verification test failed");
+#endif
+        SVI_THROW(complete_sei_nalu_and_add_to_prepend(self));
+        signing_present = 1;  // At least one SEI NALU present.
+      }
+    }
+
+  SVI_CATCH()
+  SVI_DONE(status)
+
+  free(nalu.nalu_data_wo_epb);
+
+  if (signing_present > self->signing_present) self->signing_present = signing_present;
+
+  return svi_rc_to_signed_video_rc(status);
+}
+
+SignedVideoReturnCode
+signed_video_get_sei(signed_video_t *self, uint8_t *sei, size_t *sei_size)
+{
+
+  if (!self || !sei_size) return SV_INVALID_PARAMETER;
+  *sei_size = 0;
+  if (self->num_of_completed_seis < 1) {
+    DEBUG_LOG("There are no completed seis.");
+    return SV_OK;
+  }
+  if (!sei) {
+    // Assign the SEI size to the provided pointers.
+    *sei_size = self->sei_data_buffer[0].completed_sei_size;
+    return SV_OK;
   }
-  *nalu_to_prepend = self->nalus_to_prepend_list[list_item];
-  // Memory has been transferred to the caller. Reset list item.
-  reset_nalu_to_prepend(&(self->nalus_to_prepend_list[list_item]));
+  // Assign the SEI size and SEI data to the provided pointers.
+  *sei_size = self->sei_data_buffer[0].completed_sei_size;
+  memcpy(sei, self->sei_data_buffer[0].payload, *sei_size);
+
+  // Reset the fetched SEI information from the sei buffer.
+  signed_video_nalu_data_free(self->sei_data_buffer[0].payload);
 
+  --(self->num_of_completed_seis);
+  shift_sei_buffer_index(self);
   return SV_OK;
 }
 
+static SignedVideoReturnCode
+signed_video_get_lastest_sei(signed_video_t *self, uint8_t *sei, size_t *sei_size)
+{
+  if (!self || !sei_size) return SV_INVALID_PARAMETER;
+  *sei_size = 0;
+  if (self->num_of_completed_seis < 1) {
+    DEBUG_LOG("There are no completed seis.");
+    return SV_OK;
+  }
+  if (!sei) {
+    // Assign the SEI size to the provided pointers.
+    *sei_size = self->sei_data_buffer[self->num_of_completed_seis - 1].completed_sei_size;
+    return SV_OK;
+  }
+  // Assign the SEI size and SEI data to the provided pointers.
+  *sei_size = self->sei_data_buffer[self->num_of_completed_seis - 1].completed_sei_size;
+  memcpy(sei, self->sei_data_buffer[self->num_of_completed_seis - 1].payload, *sei_size);
+
+  // Reset the fetched SEI information from the sei buffer.
+
+  --(self->num_of_completed_seis);
+  shift_sei_buffer_last_index(self);
```

Comment:
After this point the buffer has an empty item at the beginning which probably will break the system if we have more than one completed SEI. Which means that the added test is not testing things correctly.